### ISSUE TITLE
Input and output vector have different lengths, and vectors can't have named values

### ISSUE THREAD

**Issue by shiraz-shah** [sherrillmix/taxonomizr#18]
When I use accesstionToTaxa on a vector with several thousand accession numbers, the output vector that comes out is a few entries smaller in length than the input vector.

I have no chance of figuring out which accession numbers in the input caused the problem as accesstionToTaxa's output vector doesn't tell you which taxids go with which accession numbers.

A nice solution would be for accessionToTaxa and getTaxonomy to accept dataframes as input, where they append their output to the input dataframe as additional columns. It would also be an acceptable solution for both functions to accept named input vectors while preserving those names in their output vector.

Right now inputting a named vector results in the following error:
```
> head(input)
    OTU_8397    OTU_16523    OTU_10721     OTU_5863    OTU_16180     OTU_4413 
"CP013652.1" "CP016612.1" "CP002109.1" "CP022479.1" "CP011307.1" "CP022754.1" 
   OTU_16212     OTU_1206    OTU_15961     OTU_2057    OTU_14006 
"CP002403.1" "AP013044.1" "CP006594.1" "CP015971.1" "CP014282.1" 
> accessionToTaxa(input, "accessionTaxa.sql")
Error: Query and SQL mismatch
> input <- unname(input)
> head(input)
[1] "CP025590.1" "CP007443.1" "CP015399.2" "0"          "CP009872.1"
[6] "CP018970.1"
> head(accessionToTaxa(input, "accessionTaxa.sql"))
[1]    1590    1680 1834196      NA   72407     562
```

**Comment by sherrillmix** [comment]
A function taking and editing a data frame is a bit of an anti-pattern so I think I'll probably try to avoid that. 

But choking on a named vector is certainly a problem. Odd that never came up before. Thanks for catching it. I'll check on it.

```{R}
input<-c('OTU_8397' ='CP013652.1','OTU_16523'='CP016612.1','OTU_10721'='CP002109.1')
accessionToTaxa(input,'taxo/tmp/test.sql')
# Error: Query and SQL mismatch
accessionToTaxa(unname(input),'taxo/tmp/test.sql')
# [1]  162209 1130829  610130
```

Looks like it's a `if(!identical(taxaDf$accession, accessions))` which checks the input accessions to the accessions for which it is about to output IDs. If the input is a named vector then they're not "identical" even when all values are the same. A `unname` on the accessions there should fix things. I'll add that shortly. In the meantime, you could `unname` your vector when passing to `accessToTaxa`:

```{R}
accessionToTaxa(unname(input),"accessionTaxa.sql")
```

I would be pretty surprised if that generated a vector of a length different from your input since the last step in the function is the above mentioned check to make sure input and output are identical. Could you rerun with the unname and double check you're getting different length input and output? If it's still a problem, a minimal reproducible example would really help in debugging because that seems like it should not be able to pass that check without generating an error.

**Comment by shiraz-shah** [comment]
I updated the sql database and cannot for the life of me reproduce the problem. Input and output vectors are the same length now. I must have made a mistake before.

Thanks for the swift response!

**Comment by sherrillmix** [comment]
Well you definitely caught a named vector bug that I need to fix so I appreciate it in any case. Thanks and good luck with your research.

**Comment by sherrillmix** [comment]
The named vector bug is fixed in v0.6.0. Thanks for catching that.
